azurelinuxagent/ga/exthandlers.py

```diff
@@ -871,6 +927,17 @@ def collect_ext_status(self, ext):
 
         return ext_status
     
+    def is_ext_status_success(self, ext):
+        status = self.collect_ext_status(ext)
+        if status is None:
+            return (True, status)
+        if status.status != "success":
+            return (False, status)
+        for substatus in status.substatusList:
+            if substatus.status != "success":
```

Comment:
@hglkrijger - what is the interpretation of status/substatus/status... are extensions allowed to report "success" in status and, say, "error" in one of the status/substatus/status? is this considered an extension error?

Reply:
Per offline discussion @sirajudeens will double-check the validation of substatus in CRP to be sure this matches.